### PR TITLE
Allow selecting CIFS version in mount dialog

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

For https://github.com/home-assistant/supervisor/pull/4395

Add an option to select CIFS version in the mount dialog.
image
image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/ha-form/ha-form.ts [hunk 1]
```diff
@@ -106,6 +113,12 @@ export class HaForm extends LitElement implements HaFormElement {
                     ${this._computeError(error, item)}
                   </ha-alert>
                 `
+              : warning
```
Comment (ludeeus) [inline src/components/ha-form/ha-form.ts:116]: I felt showing a red error was not fitting here, so warning capabilities were added to this component.

**Comment by bramkragten** [comment]
The warning seems a bit weird, since we only show the version option if auto doesn't work (we don't know if it is because of the version I guess... Can we tell that maybe?)

Should we tell users that it didn't work and that they might have to use a different version? So a different warning message after a failure then when opened with a version number set?

**Comment by ludeeus** [comment]
Questions answered on discord.

File: src/panels/config/storage/dialog-mount-view.ts [hunk 2]
```diff
@@ -225,12 +276,29 @@ class ViewMountDialog extends LitElement {
       `ui.panel.config.storage.network_mounts.errors.${error}`
     ) || error;
 
+  private _computeWarningCallback = (warning: string): string =>
+    this.hass.localize(
+      // @ts-ignore
+      `ui.panel.config.storage.network_mounts.warnings.${warning}`
```
Comment (bramkragten) [inline src/panels/config/storage/dialog-mount-view.ts:282]: Maybe something like this?

```ts
      this._existing ? "The selected CIFS version is not recommended to use" : "We could not connect to your network share, maybe you have to change the version to make it work?"
```

Comment (ludeeus) [reply src/panels/config/storage/dialog-mount-view.ts:282]: That is 2 different things.
"We could not connect" is an error that happens after you try connecting.
The warning is for when you select an insecure version.

Comment (ludeeus) [reply src/panels/config/storage/dialog-mount-view.ts:282]: With that said, I want to change the connection error message in a new PR after this to add a button to get to the host logs.
I can also extend that message to include notes about trying a different version.

**Review by bramkragten — DISMISSED** [review]

File: src/panels/config/storage/dialog-mount-view.ts (hunk 2 above)
Comment (bramkragten) [reply src/panels/config/storage/dialog-mount-view.ts:282]: > That is 2 different things. "We could not connect" is an error that happens after you try connecting. The warning is for when you select an insecure version.

Agree it should be in the catch:
```ts
      this._validationWarning.cifs_version = "could_not_connect";
```

**Review by bramkragten — APPROVED** [review]

**Pull request merged**